.github/workflows/charm-update-libs.yaml

```
@@ -27,26 +27,86 @@ jobs:
         with:
           fetch-depth: 0
 
-      - name: Fetch charm libraries
+      - name: Merge any PRs from the automatically generated "chore/auto-libs" branch if the CI is green
+        run: |
+          # If a PR from "chore/auto-libs" is open and CI checks are passing, merge it
+          is_pr_open="$(gh pr list --head chore/auto-libs --state open --json id --jq 'length')"
+          if [[ "$is_pr_open" == "1" ]]; then
+            if gh pr checks chore/auto-libs; then
+              echo "CI checks are passing, merging the chore/auto-libs PR"
+              gh pr merge chore/auto-libs --admin --squash
+            fi
+          fi
+        env:
+          GH_TOKEN: ${{ secrets.OBSERVABILITY_NOCTUA_TOKEN }}
+
+      - name: Check for major library updates
         run: |
           sudo snap install charmcraft --classic --channel latest/stable
-          cd ${{ inputs.charm-path }}
+          cd "$GITHUB_WORKSPACE/${{ inputs.charm-path }}"
+          # Get the charm name
+          charm_name=$((yq .name metadata.yaml 2>/dev/null || yq .name charmcraft.yaml) | tr - _)
+          if [[ $charm_name = "" ]]; then echo "Error: can't extract the charm name." && exit 1; fi
+          # Initalize the issue body content to empty string, and fill it up in the for loop
+          issue_body=""
+          # For each library not belonging to the charm, check for a major version update
+          #   "lib" would be of the form `charms.prometheus_k8s.v0.prometheus_scrape`
+          for lib in $(find lib/charms/ -type f | grep '.py$' | grep -v "$charm_name" | sed 's|lib/||' | sed 's/.py//' | sed 's|/|.|g'); do
```

If we already know the charm name, how about:
```suggestion
          for lib in $(find "lib/charms/$charm_name" -type f -name "*.py" | sed 's|lib/||' | sed 's/.py//' | sed 's|/|.|g'); do
```